### ISSUE TITLE
mouse-departed not trigger if the sprite moves and the mouse doesn't

### ISSUE THREAD

**Issue by brollb** [jmoenig/Snap--Build-Your-Own-Blocks#1477]
It appears that the issue is caused by the `mouseLeave` event only being checked when the mouse moves. In this case, the morph is moving out from beneath the cursor (rather than the other way around).

http://snap.berkeley.edu/snapsource/snap.html#present:Username=brollb&ProjectName=no%20touching

**Comment by jmoenig** [comment]
Yes, that's right. ```mouseEnter``` and ```mouseLeave``` events are only triggered if the mouse moves, not if a morph moves (or rotates, or changes its scale). This is basically an optimization so we don't have to enumerate *all* morph at *every* frame. Of course, for the Snap microworld we could handle it separately, I guess.

**Comment by brianharvey** [comment]
I'm not convinced that we should change the current behavior.  If the program moves the sprite, then it already knows that it's moving away from the mouse.  The mouse event exposes something the *user* does, not something the program does.

**Comment by jmoenig** [comment]
Okay, that makes sense to me.
